tests/lib/snaps/test-snapd-service-watchdog/bin/with-helper

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+if [ -z "$NOTIFY_SOCKET" ]; then
+    echo "error: NOTIFY_SOCKET is not set"
+    exit
+fi
+
+if [ -z "$WATCHDOG_USEC" ]; then
+    echo "error: WATCHDOG_USEC is not set"
+    exit
+fi
+
+notify="systemd-notify MAINPID=$$"
```

Comment:
Not sure if it's a good idea to require systemd to be present inside the snap or its base for this to work. Might be nice to provide an implementation of the functionality on our end in snapctl, and symlink systemd-notify into it.

Let's talk further about it before investing, though.

Reply:
The future of this helper depends on whether we decide to allow notifications from all processes in cgroup or not in #4509. The current 'default' allows notifications from main process only, so any application that relies on using `systemd-notify` will not be able to work with the watchdog. Once #4509 is resolved, I'll either drop this helper or extend the spread test to cover `systemd-notify` scenario.